Python/flowgraph.c

```diff
@@ -2174,7 +2194,9 @@ push_cold_blocks_to_end(cfg_builder *g) {
             if (!IS_LABEL(b->b_next->b_label)) {
                 b->b_next->b_label.id = next_lbl++;
             }
-            basicblock_addop(explicit_jump, JUMP_NO_INTERRUPT, b->b_next->b_label.id, NO_LOCATION);
+            cfg_instr *prev_instr = basicblock_last_instr(b);
+            basicblock_addop(explicit_jump, JUMP_NO_INTERRUPT, b->b_next->b_label.id,
+                             prev_instr->i_loc);
```

Comment:
I'm going to make one more commit to make this more robust in case there are empty basic blocks (which would mean prev_instr in NULL). It doesn't seem to happen but I don't want to assume that it can't.

Reply:
Actually it can't be empty, it's the end of an exception handler so it has a POP_EXCEPT + RERAISE/RETURN. I'll add a comment and assertion.